lib/features/popular/presentation/popular_movies_page.dart

```diff
@@ -4,92 +4,92 @@ import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:movie_app/common/presentation/build_context_extensions.dart';
 import 'package:movie_app/common/presentation/widgets/app_drawer.dart';
 import 'package:movie_app/common/presentation/widgets/movie_app_bar.dart';
+import 'package:movie_app/features/popular/domain/notifiers/popular_movies_notifier.dart';
+import 'package:movie_app/features/popular/domain/providers/current_page_provider.dart';
+import 'package:movie_app/features/popular/domain/providers/total_pages_provider.dart';
+import 'package:movie_app/features/popular/presentation/widgets/popular_movies_list_widget.dart';
 import 'package:movie_app/generated/l10n.dart';
+import 'package:q_architecture/base_notifier.dart';
 
-class PopularMoviesPage extends ConsumerWidget {
+class PopularMoviesPage extends ConsumerStatefulWidget {
   static const routeName = '/popular-movies';
 
-  PopularMoviesPage({super.key});
+  const PopularMoviesPage({super.key});
+
+  @override
+  ConsumerState<PopularMoviesPage> createState() => _PopularMoviesPageState();
+}
+
+class _PopularMoviesPageState extends ConsumerState<PopularMoviesPage> {
   final GlobalKey<ScaffoldState> _globalKey = GlobalKey();
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_loadMore);
+  }
+
+  @override
+  void dispose() {
+    _scrollController.removeListener(_loadMore);
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  void _loadMore() {
+    if (_scrollController.position.pixels ==
+        _scrollController.position.maxScrollExtent) {
+      final totalPages = ref.read(totalPagesProvider);
+      final currentPage = ref.read(currentPageProvider.notifier).state;
+      final nextPage = currentPage + 1;
+
+      if (nextPage < totalPages) {
```

Comment:
Kako je ranije spomenuto, totalPages i currentPage ćeš imati u stateu od notifiera pa u biti ovdje možeš samo pozvati getMorePopularMovies ili tako nešto, a onda tamo provjeriš u stateu koji je trenutni page i da li ima još stranica i onda pozoveš getPopularMovies za sljedeću stranicu.